runtime/rasdump/dmpsup.c

```diff
@@ -577,7 +577,19 @@ configureDumpAgents(J9JavaVM *vm)
 		}
 		if ( IS_CONSUMABLE(j9vm_args, xdumpIndex) && !IS_CONSUMED(j9vm_args, xdumpIndex) )
 		{
-			GET_OPTION_VALUE(xdumpIndex, ':', &optionString);
+			/* Handle mapped tool dump options */
+			if (HAS_MAPPING(j9vm_args, xdumpIndex)) {
+				/* Construct optionString by combining the J9 tool dump command with the mapped option */
+				char *toolCursor = strstr(MAPPING_J9NAME(j9vm_args, xdumpIndex), ":tool");
+				if (toolCursor != NULL) {
+					/* The mapped option specifies the tool command to run after the equals */
+					GET_OPTION_VALUE(xdumpIndex, '=', &optionString);
+					strcat(toolCursor++, optionString);
```

Comment:
- Please don't use side-effect operators line `toolCursor++` (except in for-loops).
- check for buffer overflow